presto-native-execution/presto_cpp/main/TaskManager.cpp

```diff
@@ -746,47 +746,45 @@ std::shared_ptr<PrestoTask> TaskManager::findOrCreateTask(
 std::shared_ptr<PrestoTask> TaskManager::findOrCreateTaskLocked(
     TaskMap& taskMap,
     const TaskId& taskId) {
-  auto it = taskMap.find(taskId);
-  if (it == taskMap.cend()) {
-    auto prestoTask = std::make_shared<PrestoTask>(taskId);
-    prestoTask->info.stats.createTime =
-        util::toISOTimestamp(velox::getCurrentTimeMs());
-    prestoTask->info.needsPlan = true;
-    prestoTask->info.metadataUpdates.connectorId = "unused";
-
-    struct uuid_split {
-      int64_t lo;
-      int64_t hi;
-    };
-    union uuid_parse {
-      boost::uuids::uuid uuid;
-      uuid_split split;
-    };
-
-    uuid_parse uuid;
-    uuid.uuid = boost::uuids::random_generator()();
-
-    prestoTask->info.taskStatus.taskInstanceIdLeastSignificantBits =
-        uuid.split.lo;
-    prestoTask->info.taskStatus.taskInstanceIdMostSignificantBits =
-        uuid.split.hi;
-
-    prestoTask->info.taskStatus.state = protocol::TaskState::RUNNING;
-    prestoTask->info.taskStatus.self =
-        fmt::format("{}/v1/task/{}", baseUri_, taskId);
-    prestoTask->info.taskStatus.outputBufferUtilization = 1;
-    prestoTask->updateHeartbeatLocked();
-    ++prestoTask->info.taskStatus.version;
-
-    taskMap[taskId] = prestoTask;
-    return prestoTask;
-  } else {
-    auto prestoTask = it->second;
+  if (taskMap.count(taskId) != 0) {
```

Comment:
Here we do two searches in the map.
Can we use .find()?

Reply:
Yeah, this is not on the hot path and cost is trivial.

Reply:
Let's use count only if one look up is needed. If we need two lookups, better using find.

Reply:
Updated. Thanks!